File: JAVA DSA/BubbleSort.java
import java.util.Arrays;

public class BubbleSort {
    public static void main(String[] args) {
        int [] arrToSort = {1,2,3,4,5};
        bubbleSort(arrToSort);
    }

    static void bubbleSort(int[] arr){
        boolean swapped = false;
        for (int i = 0; i < arr.length; i++) {
            for (int j = 0; j < arr.length-i-1; j++) {
                if(arr[j] > arr[j+1]){
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                    swapped = true;
                }
            }
            if(!swapped){
                break;
            }
            System.out.println(Arrays.toString(arr));
        }
    }
}

File: SearchingAndSorting/bubbleSort.js
function bubbleSort(arr){
    for(let i = 0; i<arr.length; i++){
        for(let j = i+1; j<arr.length;j++){
            if(arr[j] > arr[j+1]){
                let temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
    console.log(arr);
}

// TC -  O(N^2)
// SC - O(N)
bubbleSort([5,4,6,3,2,1]);